src/Symfony/Bundle/FrameworkBundle/Test/MessengerAssertionsTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Component\Messenger\EventListener\MessagesSentToTransportsListener;
+use Symfony\Component\Messenger\Test\Constraint as MessengerConstraint;
+
+/*
+ * @author Marilena Ruffelaere <marilena.ruffelaere@gmail.com>
+ */
+
+trait MessengerAssertionsTrait
+{
+    /**
+     * @param int         $count   the expected number of messages
+     * @param string|null $busName The busName to consider. If null all the collected messages will be counted.
+     */
+    public static function assertMessagesByBusCount(int $count, string $busName = null, string $message = ''): void
+    {
+        self::assertThat(self::getDispatchedMessagesByBusName($busName), new MessengerConstraint\MessageCount($count, $busName), $message);
+    }
+
+    /**
+     * @param int    $count     the expected number of messages of the given class
+     * @param string $className the message object class name
+     */
+    public static function assertMessagesOfClassCount(int $count, string $className, string $message = ''): void
```

Comment:
Shouldn't it be "bus" dependent ?
Like, you have maybe 10 App\Messages\Foo in total but 3 are in bar bus and 7 in baz bus ?

And naming can be something like `assertBusAsMessageCount(string $bus, string $messageFQCN, int count, string $message='')`.